sound/soc/sof/ipc4-pcm.c

```diff
@@ -560,6 +561,23 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	case SOF_DAI_INTEL_SSP:
 		ipc4_ssp_dai_config_pcm_params_match(sdev, (char *)rtd->dai_link->name, params);
 		break;
+	case SOF_DAI_INTEL_ALH:
+		switch (ipc4_copier->frame_fmt) {
+		case SOF_IPC_FRAME_S16_LE:
+			snd_mask_none(fmt);
+			snd_mask_set_format(fmt, SNDRV_PCM_FORMAT_S16_LE);
+			break;
+		case SOF_IPC_FRAME_S24_4LE:
+			snd_mask_none(fmt);
+			snd_mask_set_format(fmt, SNDRV_PCM_FORMAT_S24_LE);
+			break;
+		case SOF_IPC_FRAME_S32_LE:
+			snd_mask_none(fmt);
+			snd_mask_set_format(fmt, SNDRV_PCM_FORMAT_S32_LE);
+			break;
+		default:
+			break;
+		}
```

Comment:
maybe you can condense  it to 
```
switch (ipc4_copier->frame_fmt) {
	case SOF_IPC_FRAME_S16_LE:
	case SOF_IPC_FRAME_S24_4LE:
	case SOF_IPC_FRAME_S32_LE:
		snd_mask_none(fmt);
		snd_mask_set_format(fmt, ipc4_copier->frame_fmt);
		break;
	default:
		break;
	}
```

Reply:
> maybe you can condense it to
> 
> ```
> switch (ipc4_copier->frame_fmt) {
> 	case SOF_IPC_FRAME_S16_LE:
> 	case SOF_IPC_FRAME_S24_4LE:
> 	case SOF_IPC_FRAME_S32_LE:
> 		snd_mask_none(fmt);
> 		snd_mask_set_format(fmt, ipc4_copier->frame_fmt);
> 		break;
> 	default:
> 		break;
> 	}
> ```

No, SOF_IPC_FRAME_S16_LE and SNDRV_PCM_FORMAT_S16_LE are not identical

Reply:
at least can use a var to store the format, then set once instead of function calling in each case.

Reply:
> at least can use a var to store the format, then set once instead of function calling in each case.

The issue is that snd_mask_none(fmt); can only be call when snd_mask_set_format() is called. So, it will look like:
```
		switch (ipc4_copier->frame_fmt) {
		case SOF_IPC_FRAME_S16_LE:
			var = SNDRV_PCM_FORMAT_S16_LE);
			break;
		case SOF_IPC_FRAME_S24_4LE:
			var = SNDRV_PCM_FORMAT_S24_LE);
			break;
		case SOF_IPC_FRAME_S32_LE:
			var = SNDRV_PCM_FORMAT_S32_LE);
			break;
		default:
			break;
		}
		if (var) {
			snd_mask_none(fmt);
			snd_mask_set_format(fmt, var);
		}
```
IMHO, I don't see any benefit on it. We add a variable and a test, but the code doesn't become simpler.